### PR TITLE
Improve sidebar presentation in specifications section

### PR DESCRIPTION
**Pull request by jeanas**
This is like https://github.com/pypa/packaging.python.org/pull/1368, for the specifications part. CC @pradyunsg

----
:books: Documentation preview :books:: https://python-packaging-user-guide--1395.org.readthedocs.build/en/1395/

### CODE REVIEW

**Review by willingc — APPROVED** [review]
Nice! Big improvement. Will leave for Pradyun and others to review too.

**Review by sinoroc — APPROVED** [review]

File: source/specifications/section-distribution-metadata.rst [hunk 1]
```diff
@@ -0,0 +1,14 @@
+=============================
+Package Distribution Metadata
+=============================
+
+.. toctree::
```
Comment (pradyunsg) [inline source/specifications/section-distribution-metadata.rst:5]:
```suggestion
.. toctree::
   :titlesonly:
```

File: source/specifications/section-distribution-formats.rst [hunk 2]
```diff
@@ -0,0 +1,8 @@
+=================================
+Package Distribution File Formats
+=================================
+
+.. toctree::
```
Comment (pradyunsg) [inline source/specifications/section-distribution-formats.rst:5]:
```suggestion
.. toctree::
   :titlesonly:
```

File: source/specifications/index.rst [hunk 3]
```diff
@@ -8,50 +8,9 @@ by the Python Packaging Authority. The process for updating these standards,
...
-   :maxdepth: 1
-
-   source-distribution-format
-   binary-distribution-format
-
-
-Package Index Interfaces
-------------------------
-
 .. toctree::
```
Comment (pradyunsg) [inline source/specifications/index.rst:11]:
```suggestion
.. toctree::
   :titlesonly:
```

File: source/specifications/section-package-indices.rst [hunk 4]
```diff
@@ -0,0 +1,8 @@
+========================
+Package Index Interfaces
+========================
+
+.. toctree::
```
Comment (pradyunsg) [inline source/specifications/section-package-indices.rst:5]:
```suggestion
.. toctree::
   :titlesonly:
```

**Review by pradyunsg — CHANGES_REQUESTED** [review]

**Comment by pradyunsg** [comment]
(let's reduce the amount of content on the "section" pages, to avoid adding too much content there)

**Review by pradyunsg — APPROVED** [review]

**Review by willingc — APPROVED** [review]

**Pull request merged**